web/app/components/base/app-icon-picker/index.tsx

```diff
@@ -69,9 +69,9 @@ const AppIconPicker: FC<AppIconPickerProps> = ({
     },
   })
 
-  const [imageCropInfo, setImageCropInfo] = useState<{ tempUrl: string; croppedAreaPixels: Area; fileName: string }>()
-  const handleImageCropped = async (tempUrl: string, croppedAreaPixels: Area, fileName: string) => {
-    setImageCropInfo({ tempUrl, croppedAreaPixels, fileName })
+  const [imageCropInfo, setImageCropInfo] = useState<{ tempUrl: string; croppedAreaPixels: Area; fileName: string; file?: File }>()
+  const handleImageCropped = async (tempUrl: string, croppedAreaPixels: Area, fileName: string, file?: File) => {
+    setImageCropInfo({ tempUrl, croppedAreaPixels, fileName, file })
   }
```

Comment:
`handleImageCropped` chould therefore rename to `handleUpload`, and `imageCropInfo` to `uploadedImageInfo`